data/no-monster/functions/kill_mobs/kill_arthropods.mcfunction

```diff
@@ -1,3 +1,14 @@
-tp @e[type=cave_spider] ~ -100 ~
-tp @e[type=silverfish] ~ -100 ~
-tp @e[type=spider] ~ -100 ~
+execute as @e[type=cave_spider] run data modify entity @s DeathLootTable set value empty
+execute as @e[type=cave_spider] run data modify entity @s Silent set value true
+# tp @e[type=cave_spider] ~ -100 ~
+kill @e[type=cave_spider]
```

Comment:
If I understand datapacks correctly, it should be possible to make a common function `kill_mob` for this:

```
data modify entity @s DeathLootTable set value empty
data modify entity @s Silent set value true
kill @s
```

... and then run it from here using the following:

```
execute as @e[type=cave_spider] run function no-monster:kill_mob
```

Aside from the obvious advantage of less duplication, this also reduces the number of times the selector is evaluated to one instead of three.

Reply:
Ooh, I'll try that one tomorrow. Didn't think of that one.